### ISSUE TITLE
[TECHDEBT] [P2P] Raintree test suite improvements: edge cases

### ISSUE THREAD

**Issue by deblasis** [pokt-network/pocket#279]
## Objective

Tend for these TODOes:

```go
./p2p/module_raintree_test.go:// IMPROVE(team): Looking into adding more tests and accounting for more edge cases.
./p2p/raintree/addrbook_utils_test.go:// IMPROVE(team): Looking into adding more tests and accounting for more edge cases.
./p2p/module_raintree_test.go:// TODO(team): make the test more rigorous but adding MaxTimes `EmitEvent` expectations. Since we are talking about more than one node

./p2p/module_raintree_test.go:	}).MaxTimes(int(expectedNumNetworkReads + 1)) // INVESTIGATE(olshansky): The +1 is necessary because there is one extra read of empty data by every channel...
```

Probably it can be tackled by looking at the code with the help of test coverage, needs confirmation so that we can update the below and eventually proceed 

## Origin Document

[ Why? Issue justification and/or link to another document]

## Goals

- _\_*REPLACE_ME*: List of things that are high level ideas or goals driving the task_
- [ ] ...

## Deliverable

- _\_*REPLACE_ME*: List of things that are concrete deliverables_
- [ ] ...

## Non-goals / Non-deliverables

- _\_*REPLACE_ME*: List of things that are out of scope_
- ...

## General issue deliverables

- [ ] Update the appropriate CHANGELOG
- [ ] Update any relevant READMEs (local and/or global)
- [ ] Update any relevant global documentation & references
- [ ] If applicable, update the source code tree explanation
- [ ] If applicable, add or update a state, sequence or flowchart diagram using [mermaid](https://mermaid-js.github.io/mermaid/)

## [Optional] Testing Methodology

- \__REPLACE_ME_: Make sure to update the testing methodology appropriately\_
- **Task specific tests**: `make ...`
- **All tests**: `make test_all`
- **LocalNet**: verify a `LocalNet` is still functioning correctly by following the instructions at [docs/development/README.md](https://github.com/pokt-network/pocket/tree/main/docs/development)

---

**Creator**: @deblasis 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
Closing this out as some of the TODOs/IMPROVEs in the code were unclear, non-actionable and outdated.

Improvements to the testing framework are being implemented in #179.

**Comment by Olshansk** [comment]
@deblasis Let's close this out once the TODOs from the code are removed.
